src/screens/MyWorkouts/MyWorkouts.tsx

```diff
@@ -25,11 +29,28 @@ const MyWorkouts: React.FC<MyWorkoutsProps> = ({ navigation }) => {
   };
 
   const updateWorkout = async (payload) => {
-    await updateEditableWorkout(payload);
+    await updateEditableWorkout(payload.id, payload);
     setEditingWorkout(undefined);
     if (refreshWorkouts) await refreshWorkouts();
   };
-
+  const deleteWorkoutBlock = async (id) => {
+    await deleteEditableWorkout(id);
+    setEditingWorkout(undefined);
+    if (refreshWorkouts) await refreshWorkouts();
+  };
+  const duplicateWorkoutBlock = async (payload: EditableWorkout) => {
+    const workout = await createEditableWorkout(session);
+    if (workout) {
+      setEditingWorkout(workout.id);
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { id, ...rest } = payload;
+      const duplicate: EditableWorkout = {
+        ...workout,
+        ...rest,
+      };
```

Comment:
Here, I suggest we use the `getEditableWorkout` function to get the data to duplicate, rather than from passing in a payload. Just seems cleaner to do all of our fetching and updating in one place rather than spread it across multiple places. Make sure you `await` the `getEditableWorkout(workoutId)` call